internal/controllers/nmc_reconciler.go

```diff
@@ -347,27 +354,32 @@ func (w *workerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1beta1.No
 
 		logger.Info("Processing worker Pod")
 
-		status := kmmv1beta1.NodeModuleStatus{
-			Namespace: modNamespace,
-			Name:      modName,
+		status := nmc.FindModuleStatus(nmcObj.Status.Modules, modNamespace, modName)
+		if status == nil {
+			status = &kmmv1beta1.NodeModuleStatus{
+				Namespace: modNamespace,
+				Name:      modName,
+			}
 		}
```

Comment:
Do we need to find the existing status? We will anyway populate it as desired in this function.

Reply:
In case the failed pod is the Unload Pod, we need to keep the existing status as is, otherwise we lose the Config, and then we cannot re-run the Unload Pod again